Replace GET with POST and update unit tests

jan-bausch opened the pull request:
This replaces calls to the Graphhopper directions API from `GET` to `POST`.

I've also updated the corresponding unit tests.

chrstnbwnkl commented on routingpy/routers/graphhopper.py:
Almost there! According to [GraphHopper docs](https://docs.graphhopper.com/#operation/postRoute)/ coordinate order is [lon, lat] for POST and the `point` parameter is pluralized. Some other parameters are pluralized as well, like `snap_prevention` and `curbsides`, so please check against the docs :)

chrstnbwnkl replied:
https://github.com/gis-ops/routingpy/blob/db5013784053e4832764845e8da8612c398cd79c/routingpy/routers/graphhopper.py#L281-L283

this is where the coordinate order needs to be changed and the `point` param needs to be changed to `points`

jan-bausch replied:
Good catch, thanks!

jan-bausch replied:
Would you like me to change the name of the function parameters as well, e.g. `snap_prevention` to `snap_preventions`? 
This would align the RoutingPy API with the GraphHopper docs. However, it would break backwards compatibility.

jan-bausch replied:
cool, I'll change that 👌

jan-bausch replied:
Done

chrstnbwnkl reviewed (changes requested):
Thanks! Just a small comment about deviating parameters between GET and POST.

jan-bausch commented:
I converted `params` into a dictionary, pluralized the relevant parameters and updated the corresponding unit test 

Needs further work, depending on https://github.com/gis-ops/routingpy/pull/91#discussion_r1141971247

jan-bausch commented:
I've updated a handful of arguments to be identical with the [docs](https://docs.graphhopper.com/#operation/postRoute):

Renamed `point_hint` to `point_hints`, `heading` to `headings`, 
`snap_prevention` to `snap_preventions`, `curb_side` to `curbsides`,

Added `custom_model` parameter

Removed `weighting`, `block_area`, `avoid`, `turn_costs` parameters

jan-bausch commented on routingpy/routers/graphhopper.py:
```diff
@@ -274,111 +258,107 @@ def directions(  # noqa: C901
         .. versionadded:: 0.3.0
            ``snap_prevention``, ``curb_side``, ``turn_costs`` parameters
+
+        .. versionchanged:: 1.2.0
+           Renamed `point_hint` to `point_hints`, `heading` to `headings`,
+           `snap_prevention` to `snap_preventions`, `curb_side` to `curbsides`,
+
+        .. versionadded:: 1.2.0
+           Added `custom_model` parameter
+
+        .. deprecated:: 1.2.0
+           Removed `weighting`, `block_area`, `avoid`, `turn_costs` parameters
```
I've just assumed a new version number `1.2.0` here. Can be changed to something else

nilsnolde replied:
Yep agreed

nilsnolde commented:
Ah damn, forgot that we have a changelog :sweat_smile: We'll get it before we release (likely this week)

jan-bausch commented:
Awesome, thanks for the quick feedback 🎉
